### PR TITLE
Set alias @bower and @npm to @vendor/{bower-asset,@npm-asset}

### LINKED ISSUE #152: Update assets aliases
This issue has originally been reported by @chervand at https://github.com/yiisoft/yii2/issues/15088.
Moved here by @cebe.

-----

Since switched to `asset-packagist.org`, `yii\base\Application::setVendorPath()` `bower` and `npm` paths should be updated `bower-asset` and `npm-asset` respectively.

### What steps will reproduce the problem?

Using path aliases such as `@bower` or `@npm`. For example, `yiisoft/yii2-bootstrap` uses `@bower` alias for assets.

### What is the expected result?

 `@bower` and `@npm` are aliases for `@vendor/bower-asset` and `@vendor/npm-asset` respectively.

### What do you get instead?

Wrong paths for `bower` and `npm` assets while using its path aliases.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.13
| PHP version      | 7.0.22
| Operating system | ubuntu0.16.04.1

### PR DESCRIPTION
**Pull request by ricpelo**
Fixes #152.

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #152

### CODE REVIEW

File: apidoc [hunk 1]
```diff
@@ -50,5 +50,7 @@ $application = new yii\console\Application([
 if ($vendorPath !== null) {
     $application->setVendorPath($vendorPath);
 }
+Yii::setAlias('@bower', '@vendor/bower-asset');
```
Comment (chervand) [inline apidoc]: I think, it should be better set via config array.

Comment (chervand) [reply apidoc]: @ricpelo aha, I see, setVendorPath() overrides it anyway

Comment (cebe) [reply apidoc]: this breaks it for all cases where its a different directory name. I think its better to go the same way as with composer above and check whether either `bower` or `bower-asset` exist and use the one that is available.

File: apidoc [hunk 2]
```diff
@@ -50,5 +50,33 @@ $application = new yii\console\Application([
...
+    __DIR__ . '../../npm-asset',
+    __DIR__ . '../../../../npm',
+    __DIR__ . '../../../../npm-asset',
+];
+foreach ($npmDirs as $dir) {
+    if (file_exists($dir)) {
+        Yii::setAlias('@npm', $dir);
+        break;
+    }
+}
```
Comment (ricpelo) [inline apidoc]: I've pushed a new version. What about this, @cebe? Should I use another set of directories? Thanks!

File: apidoc [hunk 3]
```diff
@@ -51,12 +51,8 @@ if ($vendorPath !== null) {
...
 }
 $bowerDirs = [
-    __DIR__ . '/vendor/bower',
-    __DIR__ . '/vendor/bower-asset',
-    __DIR__ . '../../bower',
-    __DIR__ . '../../bower-asset',
-    __DIR__ . '../../../../bower',
-    __DIR__ . '../../../../bower-asset',
+    "$vendorPath/bower",
+    "$vendorPath/bower-asset",
```
Comment (ricpelo) [inline apidoc]: Hope this helps.

File: apidoc (hunk 2 above)
Comment (cebe) [reply apidoc]: looks great, however I just noticed that `$vendorPath` could possibly be null, not sure what the use case is but your code should go into the if statement above that checks for `if ($vendorPath !== null) {`. Its ready to merge after that. Thanks!

Comment (ricpelo) [reply apidoc]: Ok!

Comment (ricpelo) [reply apidoc]: Looks good now?

**Comment by cebe** [comment]
Thank you!

**Comment by hujuice** [comment]
Is that enough?
I'm still running in:
`Rendering files: 0% (0/39) ETA: n/a   Error: The file or directory to be published does not exist: /path/to/my/project/vendor/yiisoft/yii2-apidoc/../../bower/bootstrap/dist`
I'm unable to quickly debug that, but I always see 'bower' (and not 'bower-asset') in the 'extra' section of the composer.json.

**Comment by cebe** [comment]
are you sure you are running a version with this patch applied? it should work with `bower` as well as `bower-asset` now.

**Comment by hujuice** [comment]
You are right.
Running 'composer require --dev "yiisoft/yii2-apidoc": "dev-master#96e1651" resolved.
I'm very sorry.

**Comment by cebe** [comment]
no need to be sorry, I'm happy to help ;)

**Pull request merged**